src/Persistence/PersistentObjectFactory.php

```diff
@@ -372,26 +380,56 @@ protected function normalizeCollection(string $field, FactoryCollection $collect
      *
      * @internal
      */
-    protected function normalizeObject(object $object): object
+    protected function normalizeObject(string $field, object $object): object
     {
         $configuration = Configuration::instance();
 
-        if (
-            !$this->isPersisting()
-            || !$configuration->isPersistenceAvailable()
-        ) {
+        $object = unproxy($object, withAutoRefresh: false);
+
+        if (!$configuration->isPersistenceAvailable()) {
             return $object;
         }
 
-        $object = unproxy($object, withAutoRefresh: false);
-
         $persistenceManager = $configuration->persistence();
+
         if (!$persistenceManager->hasPersistenceFor($object)) {
             return $object;
         }
 
+        $inverseRelationship = $persistenceManager->bidirectionalRelationshipMetadata(static::class(), $object::class, $field);
```

Comment:
I really would like to simplify this, or to hide the complexity, but I don't know how :shrug: